### ISSUE TITLE
Can I use this repo to upload multiple files to one web page?

### ISSUE THREAD

**Issue by zydjohnHotmail** [Nikms07/multiple-file-uploader#1]
Hello:
I created one facebook page, and I can use creator studio to upload multiple files at one shot. If you upload files one by one, then quite often you can upload only a few files (less than 5), but if you use creator studio, you can upload more than 50 files at one shot. I tested with 48 files in one shot, and it works.
I want to know if I can use this repo to upload a number of files inside one folder, like: [C:\Uploads](https://github.com/Uploads).
To make things simple, if I put 2 files in the folder: C:\Uploads1.mp4 and C:\Uploads\2.mp4. 
Inside my facebook page, I can go to this URL to access creator studio: 
https://business.facebook.com/creatorstudio/home?reference=search_facebook
Then click on upload button and search for the folder where the files to be uploaded are saved, then Ctrl+A to select them all, then Ctrl+V to paste them in creator studio, then all the files will be uploaded.
Let me know know how I can use this repo to upload the 2 files in this folder to my facebook page.
If you have gamil account, you can register facebook without any issue, and creating a facebook is also very easy, if you don’t have mp4 format files, then you can upload other format files, like: JPG, PNG, GIF, MP3 etc..
Please advise,
Thanks,

**Comment by Nikms07** [comment]
Hi.

Not very sure if I have clearly understood your question, like where exactly you want to upload the content. ok I am guessing you to want to upload pictures in a folder i.e. uploads. 

Right now this repo is posting content to an AWS s3 bucket but if you want to upload content into a folder on the server only you can surely do that. you can see my initial/previous commits of this repo where it is shown how to upload content in uploads folder. 

let me know if you were able to do it or contact me again if you need further assistance. 

Thanks

**Comment by zydjohnHotmail** [comment]
Hello:
This is very very specific question. To understand my question, you have to first have a facebook account, if you already have a gmail/google account, then simply sign in facebook, you will get your facebook account.  Then you can create one page of your own, don't worry, this is also very simple, takes about 5 to 10 minutes.  Then you may understand my issue: there is one business tools in facebook called "Creator Studio", it can upload up to 50 files in one shot.  But you don't have to upload 50 files, it is best that you get 2 files ready for upload, any format like: JPG, PNG, PDF, MP4, MP3, WAV or even TXT format.  Then you can use "Creator Studio" to upload 2 files.  My issue is I want to use API or other automatic method to upload at least 2 files in "Creator Studio".  Something is not easy to understand if you don't have the platform.
For example, if I have some issue on how to use Microsoft Visual Studio 2022, then I will not post any question for repo mainly for Java IDE like Eclipse.

**Comment by Nikms07** [comment]
Hi

So as you said i now have a facebook account and page. I have seen creator
studio and uploading thing where you can upload up to 50 videos by going to
the "Multiple upload" button and i can select the videos i want to upload
and it will be uploaded.

Still I didn't get the point actually where in this we need this repo
actually, like you can just upload the videos from your pc just by browsing
them.

Why we need an extra layer in between like An API or something like that?

Please help me to understand this.

Thanks

**Comment by zydjohnHotmail** [comment]
Hello:
Good to hear that you see the "Creator Studio" in Facebook.  In my case, I always have 50 to 100 files to upload every day, then to upload each file and provide a name and description is not as easy as if you want to upload only 2 files.
If you try to do this for 50 files: 4 fields of text for input of each file, 200 input text for all in order to upload them in one shot.
File name: 1.mp4; Post title: title for 1.mp4; Post description: description for 1.mp4; Tag for 1.mp4; 
File name: 2.mp4; Post title: title for 2.mp4; Post description: description for 2.mp4; Tag for 2.mp4; 
......
File name: 50.mp4; Post title: title for 50.mp4; Post description: description for 50.mp4; Tag for 50.mp4; 
I have information for all 50/100 files in database, however, if you want to type all the above information by hand, can you do this without making typos?
If I can use the repo with API code, then I can make sure all 200 input text records are correct.
Hope you understand, if there is only 1 or 2 files to upload, I already have code in Playwright, but if I upload file one by one, then usually, I reached my upload limit after around 5 files, but if I wait until  the second day to upload more files, then I will have more files (50 to 100 more on the second day) to upload.

**Comment by Nikms07** [comment]
Hi

So as I understood, yuo want to upload the files in creator studio using this repo like I am uploading files in AWS S3 bucket using multer and AWS API. 

Please correct me if I am wrong. 

Now in your case, you have to work with multer and facebook API. 

You can upload the files with input fields from your database through multer to server (in my case nodejs server) and from there you have to see how you can upload those files from server to creator studio using facebook API. 

But i am not very sure how all the input fields will go through with multer in one go because with multer you can send multiple files but for other input fields you will have to create a input tag for every field and it will be the same problem that you are facing right now, which is essentially typing all the information by hand. 

I don't think this will be a right solution to your problem, Instead I think you should look for a solution which directly works with facebook API. Like you should be able to upload the videos directly by calling facebook API.  
[See if this helps you.](https://developers.facebook.com/docs/video-api/guides/publishing)

Thanks.

**Comment by Nikms07** [comment]
Hi

So as I have understood, you want to upload the files in creator studio
using this repo like I am uploading files in AWS s3 bucket using multer and
AWS API.

Please correct me if I am wrong.

Now in your case , you have to work with multer and facebook API.

You can upload the files with input fields from your database thorugh
multer to server (in my case nodejs server)  and from there you have to see
how you can upload those files from server to creator studio using facebook
API .

Thanks.

**Comment by zydjohnHotmail** [comment]
Hello:
Thanks for your advice, but I figured it out.
I think facebook has been chaning the web page strcture recently, so it is not easy to know how to use their API or other stuff.
Anyway, I found that if I switch the page view to the old view, then I can still use code in Playwright to upload multiple files (I succeed to upload 100+ videos in one short).
Therefore, in stead of coding for new solution, I simply switch to the old page view, then run my old code, then it works!
